src/main/scala/cromwell/binding/command/ParameterCommandPart.scala

```diff
@@ -3,17 +3,19 @@ package cromwell.binding.command
 import cromwell.binding.WdlValue
 import cromwell.binding.types.WdlType
 
+import scala.util.Try
+
 case class ParameterCommandPart(wdlType: WdlType, name: String) extends CommandPart {
   override def toString: String = "${" + s"$wdlType $name" + "}"
 
-  def instantiate(parameters: Map[String, WdlValue]): String = {
-    val paramValue = parameters.getOrElse(name, {
-      throw new UnsupportedOperationException(s"Parameter $name not found")
-    })
-    if (wdlType != paramValue.wdlType) {
-      throw new UnsupportedOperationException(s"Incompatible type for $name: need a $wdlType, got a ${paramValue.wdlType}")
-    }
-    /* TODO: this also is a gross over-simplification */
-    paramValue.value.toString
-  }
+  def instantiate(parameters: Map[String, WdlValue]): Try[String] =
```

Comment:
Cue the Try vs. try discussion...

First of all, thanks for laying out a concrete example, this helps me a lot to understand it.

My opinion so far is that I can see some merit, but it seems to not be adding a whole lot for us.  I get that now it doesn't bail at the first error and we could collect all the errors that happened, but that seems only marginally useful and we pay a bit of a price in complexity (see changes in `Command.scala`).  Is there anyway we could get around that code in `Command.scala`?  I 

The `Try[]` approach also doesn't really save us from catching and handling an exception, it just changes the syntax from `try... catch...` to a pattern match.